### PR TITLE
Quantized conv avoid functional usage

### PR DESCRIPTION
**Pull request by z-a-f**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#22733 Quantized conv avoid functional usage**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15835572/)

This refactor changes the conv module to avoid the usage of the functional ops.

Differential Revision: [D15835572](https://our.internmc.facebook.com/intern/diff/D15835572/)

### CODE REVIEW

**Comment by zafartahirov** [comment]
Something went terribly wrong. THe old patch (#21808) got corrupted -- created this one.

File: test/test_nn_quantized.py [hunk 1]
```diff
@@ -149,7 +148,7 @@ def test_conv_api(self):
         self.assertTrue(hasattr(conv_under_test, '_zero_point'))
 
         # Test properties
-        # self.assertEqual(qw, conv_under_test.weight)
```
Comment (jerryzh168) [inline test/test_nn_quantized.py]: remove?

File: torch/nn/quantized/modules/conv.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 # coding=utf-8
+r"""Quantized convolution modules."""
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: why do we need this comment?

File: torch/nn/quantized/modules/conv.py [hunk 3]
```diff
@@ -52,15 +54,14 @@ def __init__(self, in_channels, out_channels, kernel_size, stride=1,
         qbias = torch._empty_affine_quantized([out_channels],
                                               scale=1, zero_point=0,
                                               dtype=torch.qint32)
-        self.register_buffer('_packed_weight',
-            torch.ops.quantized.fbgemm_conv_prepack(qweight, self.groups))
+        self.register_buffer('_packed_weight', torch.ops.quantized.fbgemm_conv_prepack(qweight, self.groups))
         self.register_buffer('bias', qbias)
         self.register_buffer('_scale', torch.tensor([1.0], dtype=torch.double))
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: can you change the setter getter of scale, zero_point as well.

**Review by jerryzh168 — APPROVED** [review]
minor comments

File: torch/nn/quantized/modules/conv.py (hunk 2 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/conv.py]: These are "package-level docstrings". Ideally we want to have them in all importable files. That way, the user can run something like

```python
from torch.nn.quantized.modules import conv
help(conv)
```

And that would present the docstring. Ideally we want to have more info in there. If this is not the pattern we are following, I can remove it.

File: torch/nn/quantized/modules/conv.py (hunk 3 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/conv.py]: You mean setters to explicitly set the tensor dtype? Will do!

File: torch/nn/quantized/modules/conv.py [hunk 4]
```diff
@@ -75,7 +77,7 @@ def scale(self, s):
         if isinstance(s, torch.Tensor):
             self._scale = s
         else:
-            self._scale = torch.Tensor([s])
+            self._scale = torch.tensor([s], dtype=torch.double)
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/conv.py]: Can we ensure that this is consistent across all quantized modules? @zafartahirov : Can you verify this?

**Review by raghuramank100 — APPROVED** [review]
Have a request for a follow up task, please address

File: torch/nn/quantized/modules/conv.py (hunk 4 above)
Comment (zafartahirov) [reply torch/nn/quantized/modules/conv.py]: I can go through the modules to make sure this is the case. However, setting the tensor to double is optional (I think) -- it is the default.

**Pull request closed without merging**